src/iter/test.rs

```diff
@@ -1722,3 +1722,33 @@ fn check_either_extend() {
     right.par_extend(v.clone());
     assert_eq!(right, Either::Right(v.iter().cloned().collect()));
 }
+
+#[test]
+fn check_interleave_eq() {
+    let xs: Vec<usize> = (0..10).collect();
+    let ys: Vec<usize> = (10..20).collect();
+
+    let mut actual = vec![];
+    xs.par_iter().interleave(&ys).map(|&i| i).collect_into(&mut actual);
+
+    let expected: Vec<usize> = (0..10).zip((10..20)).flat_map(|(i, j)| vec![i, j].into_iter()).collect();
+    assert_eq!(expected, actual);
+}
+
+#[test]
+fn check_interleave_uneven() {
+
+    let cases: Vec<(Vec<usize>, Vec<usize>, Vec<usize>)> = vec![
+        ((0..9).collect(), vec![10], vec![0, 10, 1, 2, 3, 4, 5, 6, 7, 8]),
+        (vec![10], (0..9).collect(), vec![10, 0, 1, 2, 3, 4, 5, 6, 7, 8]),
+        ((0..5).collect(), (5..10).collect(), (0..5).zip((5..10)).flat_map(|(i, j)| vec![i, j].into_iter()).collect()),
+        (vec![], (0..9).collect(), (0..9).collect()),
+        ((0..9).collect(), vec![], (0..9).collect())
+    ];
+
+    for (i, (xs, ys, expected)) in cases.into_iter().enumerate() {
+        let mut res = vec![];
+        xs.par_iter().interleave(&ys).map(|&i| i).collect_into(&mut res);
+        assert_eq!(expected, res, "Case {} failed", i+1);
+    }
+}
```

Comment:
How about a test under `rev()`, for the `next_back` behavior?

Reply:
I added another `assert_eq` that tests each case when using `rev()`